pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/GrpcMailboxService.java

```diff
@@ -42,22 +44,50 @@
  *   to open 2 mailboxes, they should use {job_id}_1 and {job_id}_2 to distinguish the 2 different mailbox.</li>
  * </ul>
  */
-public class GrpcMailboxService implements MailboxService<MailboxContent> {
+public class GrpcMailboxService implements MailboxService<BaseDataBlock> {
   // channel manager
+  private static final int CAPACITY = 100;
   private final ChannelManager _channelManager;
   private final String _hostname;
   private final int _mailboxPort;
 
   // maintaining a list of registered mailboxes.
-  private final ConcurrentHashMap<String, ReceivingMailbox<MailboxContent>> _receivingMailboxMap =
+  private final ArrayBlockingQueue<CleanableEntity> _mailboxQueue = new ArrayBlockingQueue<>(10000);
+  private final ArrayBlockingQueue<CleanableEntity> _channelQueue = new ArrayBlockingQueue<>(10000);
+  private final ConcurrentHashMap<String, ReceivingMailbox<BaseDataBlock>> _receivingMailboxMap =
       new ConcurrentHashMap<>();
-  private final ConcurrentHashMap<String, SendingMailbox<MailboxContent>> _sendingMailboxMap =
+  private final ConcurrentHashMap<String, SendingMailbox<BaseDataBlock>> _sendingMailboxMap =
+      new ConcurrentHashMap<>();
+  private final ConcurrentHashMap<String, PassThroughChannel<BaseDataBlock>> _channelMap =
       new ConcurrentHashMap<>();
 
   public GrpcMailboxService(String hostname, int mailboxPort, PinotConfiguration extraConfig) {
     _hostname = hostname;
     _mailboxPort = mailboxPort;
     _channelManager = new ChannelManager(this, extraConfig);
+    Thread t = new Thread(() -> {
```

Comment:
This is again a hack to fix the memory leak in the maps in GrpcMailboxService. This leak also exists for the current open-source join implementation (lmk if it's already fixed).